### ISSUE TITLE
Exporting data in YOLO format

### ISSUE THREAD

**Issue by amitguptagwl** [NaturalIntelligence/imglab#83]
Currently imglab supports exporting labelled data in Pascal VOC, COCO format. We want to let users export data in YOLO format as well.

target folder : [data formaters](https://github.com/NaturalIntelligence/imglab/blob/master/dataformaters)

Please check how export feature is implemented for other formats in  [data formaters](https://github.com/NaturalIntelligence/imglab/blob/master/dataformaters)

> [Watch](https://github.com/NaturalIntelligence/imglab/watchers) for changes, or [Bookmark](https://github.com/NaturalIntelligence/imglab/stargazers) for easy discovery.
> [Fund](https://www.patreon.com/bePatron?u=9531404) this project for new features and maintenance.
> [Showcase] your project with us by raising an issue

**Comment by JacobMarkBaird1998** [comment]
Hi I am a newcomer and would like to contribute to this project, which would be my first. Has this task been assigned yet?

**Comment by amitguptagwl** [comment]
@JacobMarkBaird1998 Yes you can go ahead with this. Let me know if you need any help.

**Comment by JacobMarkBaird1998** [comment]
Thank you, @amitguptagwl . I have started working on this feature but when I try to run the index.html file saved in my local repository, it seems I am missing some lots of components. I have cloned the entire repository from GitHub so there should be no missing files, so I can't figure out what is wrong with it. Do you have any idea?
Thanks.
Here is a screenshot of what appears when I open index.html in Google Chrome:
![image](https://user-images.githubusercontent.com/42558631/44872597-5b1ef100-ac6c-11e8-959c-a5a892ffacba.png)

**Comment by JacobMarkBaird1998** [comment]
Thank you, I've been looking for resources to familiarize myself with the YOLO format. I've found resources that say the file extension is just .txt. Does this sound right? I'm using these pages as reference:
https://github.com/AlexeyAB/Yolo_mark/issues/60
http://guanghan.info/download/before_conversion.txt
http://guanghan.info/download/after_conversion.txt

**Comment by amitguptagwl** [comment]
It looks good. 
You can generate YOLO file from other tools as well and compare if both are same. https://github.com/tzutalin/labelImg

You can also check this python code for implementation reference: https://github.com/tzutalin/labelImg/blob/master/libs/yolo_io.py

Finally you can import generated file in other image annotation application to test.

**Comment by JacobMarkBaird1998** [comment]
Thanks a lot Amit!

**Comment by amitguptagwl** [comment]
The image which is currently open should be written in YOLO format. However it'll good if we can ask user to "export current" or "export all".

For this we'll have to redesign export pop up window.

**Comment by JacobMarkBaird1998** [comment]
Okay, thank you. This is what I was going with until now.

**Comment by JacobMarkBaird1998** [comment]
I am getting this error when trying to call my "yoloFormater.toYOLO()" function defined inside my yolo.js file. Any idea what the issue is, and how I might solve it?
Thanks
![image](https://user-images.githubusercontent.com/42558631/44946808-b3273600-add9-11e8-8eda-ebca1e3a0425.png)
![image](https://user-images.githubusercontent.com/42558631/44946814-cafeba00-add9-11e8-902e-532f439814d1.png)
![image](https://user-images.githubusercontent.com/42558631/44946819-d6ea7c00-add9-11e8-97f0-9574828dac8b.png)

**Comment by amitguptagwl** [comment]
I believe you're not including your JS file in index.html

**Comment by amitguptagwl** [comment]
Have you tested that the generated file is working fine other application too?

You need to fork to submit the PR. You can't directly push in this repository. 

Once your PR is successfully merged, you can submit it for the rewards on steemit.com which will be reviewed by utopian. So you need take care of it's tag, and format. Check this submission : 
 https://steemit.com/utopian-io/@alvinvoo/project-update-imglab-new-features

**Comment by JacobMarkBaird1998** [comment]
Thank you, Amit. Sorry for the easy questions, I'm just a newbie to GitHub and haven't used it outside of school at all haha

**Comment by amitguptagwl** [comment]
No problem. My main aim of creating issues is to give the chance to new contributors and mentor them. Please check this link: https://github.com/Roshanjossey/first-contributions

**Comment by amitguptagwl** [comment]
This issue is ready to be picked by any contributor. You can check #97 for implementation help and other discussions.

**Comment by kenrubiooo** [comment]
> I am getting this error when trying to call my "yoloFormater.toYOLO()" function defined inside my yolo.js file. Any idea what the issue is, and how I might solve it?
> Thanks
> ![image](https://user-images.githubusercontent.com/42558631/44946808-b3273600-add9-11e8-8eda-ebca1e3a0425.png)
> ![image](https://user-images.githubusercontent.com/42558631/44946814-cafeba00-add9-11e8-902e-532f439814d1.png)
> ![image](https://user-images.githubusercontent.com/42558631/44946819-d6ea7c00-add9-11e8-97f0-9574828dac8b.png)

Hi @JacobMarkBaird1998 

Is dw = 1/image_width and dh = 1/image_height?
Also, how do you get shape.bbox.x and shape.bbox.y?

I wanted to convert the float form of x,y,w,h from the yolo format to a format like 240, 240, 100, 150 where (240, 240) are centers of the boxes and (100,150) are the width and the height of the boxes. Thank you!

**Comment by amitguptagwl** [comment]
@kenrubiooo  You're trying to open the html file directly in the browser. Try http-server or live-server to run the app locally to test. Check this [guide](https://github.com/NaturalIntelligence/imglab/blob/master/docs/guide.md#offline-installation) for more detail.
